lib/lambda.js

```diff
@@ -9,6 +9,32 @@ const path = require('path')
 let isFirstRun = true
 let gFaasId // Set on first invocation.
 
+// The trigger types for which we support special handling.
+// https://docs.aws.amazon.com/lambda/latest/dg/lambda-services.html
+const TRIGGER_GENERIC = 1
```

Comment:
I see we've replaced the smaller functions that identify an individual payload type (ex. `isApiGatewayEvent`) with this larger/more-complex function `triggerTypeFromEvent` function that identifies an individual payload type, and then we call `triggerTypeFromEvent` once its result value is passed around where it needs to be.  The main tradeoffs here seem to be 

- the perf. of multiple calls to the older small function vs. the higher memory use of keeping the constant-ish `TRIGGER_` variables around. 

- Swapping where the complexity goes :) 

No objections, just noting the change out loud.